cannot build  from source with gcc-9.2.1 (on Fedora)

When I follow the instructions to build from source (on Fedora), I get
```
cmake ../../src -DCMAKE_INSTALL_PREFIX=/opt/lean/v3.4.2 -DCMAKE_BUILD_TYPE=RELEASE
make -j
...
Scanning dependencies of target standard_lib
/home/waldmann/software/prover/lean/library/init/meta/tactic.lean: parsing at line 1399terminate called after throwing an instance of 'lean::exception'
  what():  vm check failed: is_composite(o) (possibly due to incorrect axioms, or sorry)
make[2]: *** [CMakeFiles/standard_lib.dir/build.make:57: CMakeFiles/standard_lib] Aborted (core dumped)
make[1]: *** [CMakeFiles/Makefile2:954: CMakeFiles/standard_lib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```
most of the time. This happen with  `gcc (GCC) 9.2.1 20190827 (Red Hat 9.2.1-1)`

The build works with
* `gcc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0`
* `clang version 9.0.0 (Fedora 9.0.0-1.fc31)`
* `g++ (GCC) 9.2.0`  (built from source) (on Fedora)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -238,8 +238,9 @@ endif()
 
 # Compiler-specific C++11 activation.
 if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
+    # Both -dump* arguments are needed, see https://stackoverflow.com/a/47410010
     execute_process(
-        COMMAND "${CMAKE_CXX_COMPILER}" -dumpversion OUTPUT_VARIABLE GCC_VERSION)
+        COMMAND "${CMAKE_CXX_COMPILER}" -dumpfullversion -dumpversion OUTPUT_VARIABLE GCC_VERSION)
     if (NOT (GCC_VERSION VERSION_GREATER 4.9 OR GCC_VERSION VERSION_EQUAL 4.9))
         message(FATAL_ERROR "${PROJECT_NAME} requires g++ 4.9 or greater.")
     # TODO(Vtec234): Add upper bound when GCC with a fix is released

